src/digest/digest.rs

```diff
@@ -153,35 +153,43 @@ impl Context {
         padding_pos += 1;
 
         if padding_pos > self.algorithm.block_len - self.algorithm.len_len {
-            polyfill::slice::fill(
-                &mut self.pending[padding_pos..self.algorithm.block_len], 0);
+            polyfill::slice::fill(&mut self.pending[padding_pos..self.algorithm
+                                      .block_len],
+                                  0);
             unsafe {
                 (self.algorithm.block_data_order)(&mut self.state,
-                                                  self.pending.as_ptr(), 1);
+                                                  self.pending.as_ptr(),
+                                                  1);
             }
             // We don't increase |self.completed_data_blocks| because the
             // padding isn't data, and so it isn't included in the data length.
             padding_pos = 0;
         }
 
-        polyfill::slice::fill(
-            &mut self.pending[padding_pos..(self.algorithm.block_len - 8)], 0);
+        let slice = self.pending[padding_pos..(self.algorithm.block_len - 8)];
```

Comment:
the inline calculation in the slice confused rustfmt mightily so I pulled out a local